src/main/java/delight/nashornsandbox/internal/NashornDetection.java

```diff
@@ -0,0 +1,92 @@
+package delight.nashornsandbox.internal;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class NashornDetection {
+
+    private static final Logger logger = LoggerFactory.getLogger(NashornDetection.class);
+
+    private static final Class<?> JDK_NASHORN_ScriptObjectMirror_CLASS;
+    private static final Class<?> JDK_NASHORN_NashornScriptEngineFactory_CLASS;
+    private static final Class<?> JDK_NASHORN_ClassFilter_CLASS;
+
+    private static final Class<?> STANDALONE_NASHORN_ScriptObjectMirror_CLASS;
+    private static final Class<?> STANDALONE_NASHORN_NashornScriptEngineFactory_CLASS;
+    private static final Class<?> STANDALONE_NASHORN_ClassFilter_CLASS;
+
+    static {
+        JDK_NASHORN_ScriptObjectMirror_CLASS = findClass("jdk.nashorn.api.scripting.ScriptObjectMirror", "JDK-provided Nashorn not found");
+        if (JDK_NASHORN_ScriptObjectMirror_CLASS != null) {
+            JDK_NASHORN_NashornScriptEngineFactory_CLASS = findClass("jdk.nashorn.api.scripting.NashornScriptEngineFactory", "");
+            JDK_NASHORN_ClassFilter_CLASS = findClass("jdk.nashorn.api.scripting.ClassFilter", "");
+        } else {
+            // no need to search for those and add more logs
+            JDK_NASHORN_NashornScriptEngineFactory_CLASS = null;
+            JDK_NASHORN_ClassFilter_CLASS = null;
+        }
+
+        STANDALONE_NASHORN_ScriptObjectMirror_CLASS = findClass("org.openjdk.nashorn.api.scripting.ScriptObjectMirror", "Standalone Nashorn not found");
+        if (STANDALONE_NASHORN_ScriptObjectMirror_CLASS == null) {
+            STANDALONE_NASHORN_NashornScriptEngineFactory_CLASS = findClass("org.openjdk.nashorn.api.scripting.NashornScriptEngineFactory", "");
+            STANDALONE_NASHORN_ClassFilter_CLASS = findClass("org.openjdk.nashorn.api.scripting.ClassFilter", "");
+        } else {
+            STANDALONE_NASHORN_NashornScriptEngineFactory_CLASS = null;
+            STANDALONE_NASHORN_ClassFilter_CLASS = null;
+        }
+        // TODO add a report of what was detected and what will be used
```

Comment:
Yep I think so, as you said maybe a debug level log?

Reply:
yes, I had a sleepless night so I worked on this a bit, but it's not finished yet

Reply:
One step at a time, and take care!